utils/minio_utils.py

```
@@ -0,0 +1,163 @@
+            file_size = file_data.seek(0, 2)
+            file_data.seek(0)
+
+            self.client.put_object(
+                bucket_name=self.bucket_name,
+                object_name=object_name,
+                data=file_data,
+                length=file_size,
+                content_type=content_type,
+            )
+        except Exception as e:
+            raise Exception(f"Failed to upload {object_name}: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)